### PR TITLE
Bugfix: Exception in split method not working as intented

### PR DESCRIPTION
**Pull request by yannick-mayeur**
Split method testing if ">" but should test if ">=".
Added a test for this exception.

Signed-off-by: Yannick Mayeur <yannick.mayeur@gmail.com>

### CODE REVIEW

File: org.eclipse.january.test/src/org/eclipse/january/dataset/DatasetUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.january.dataset;
+
+import org.junit.*;
```
Comment (jonahkichwacoders) [inline org.eclipse.january.test/src/org/eclipse/january/dataset/DatasetUtilsTest.java]: Run Organize Imports (Ctrl-Shift-o)  on this to remove the .* see http://javadude.com/articles/importondemandisevil.html for more info

File: org.eclipse.january/src/org/eclipse/january/dataset/DatasetUtils.java [hunk 2]
```diff
@@ -476,7 +476,7 @@ public static Dataset concatenate(final IDataset[] as, final int axis) {
 	public static List<Dataset> split(final Dataset a, int[] indices, final int axis) {
 		final int[] ashape = a.getShapeRef();
 		final int rank = ashape.length;
-		if (axis > rank) {
+		if (axis >= rank) {
```
Comment (jonahkichwacoders) [inline org.eclipse.january/src/org/eclipse/january/dataset/DatasetUtils.java]: I see a few lines above in the other split there is apparently the same bug, please fix it there too. Also have a quick grep/find in files of the code to see if this copy-paste error exists elsewhere.

Comment (yannick-mayeur) [inline org.eclipse.january/src/org/eclipse/january/dataset/DatasetUtils.java]: done

File: org.eclipse.january.test/src/org/eclipse/january/dataset/DatasetUtilsTest.java (hunk 1 above)
Comment (yannick-mayeur) [inline org.eclipse.january.test/src/org/eclipse/january/dataset/DatasetUtilsTest.java]: done

**Comment by jonahkichwacoders** [comment]
Waiting on travis, once it is done this is ready.

**Pull request merged**